### ISSUE TITLE
`set virtualedit=all` doesn't works in normal-mode

### ISSUE THREAD

**Issue by unclechu** [bryphe/oni#399]
1. In *normal mode* when I'm trying to move cursor after last filled column it immediately returns back;
2. In *insert mode* it acts the same when auto-complete window is opened;
3. In *insert mode* after auto-complete window was opened and I move cursor by lines it doesn't close but may be it should be separated issue.

[Short screencast that demonstrates it](https://youtu.be/r_bsYfBRcPE).

**Comment by extr0py** [comment]
Interesting, didn't know about `virtualedit` until now 😄 . Seems like it isn't working at all in Oni.

I was curious if this was specific to Oni, and tried it out in neovim-qt.exe, which is working fine. Curious to see why this isn't working in Oni - I wonder if there is a conflict with some of the built-in settings.

**Comment by unclechu** [comment]
@extr0py you could look at help info for `getcurpos()` vim function:
```
getcurpos()	Get the position of the cursor.  This is like getpos('.'), but
		includes an extra item in the list:
		    [bufnum, lnum, col, off, curswant] ~
		The "curswant" number is the preferred column when moving the
		cursor vertically.
		This can be used to save and restore the cursor position: >
			let save_cursor = getcurpos()
			MoveTheCursorAround
			call setpos('.', save_cursor)
```
May be the Oni uses this function to get cursor position data and uses `col` to get cursor column position but it should use `curswant` instead of.

**Comment by Uroc327** [comment]
This seems to be fixed (as well as #447 ) . `:set virtualedit=all` seems to result in the desired behavior: free cursor movements in all modes.

**Comment by bryphe** [comment]
Excellent, I'll close this out too.
